app/screens/edit_profile/edit_profile.test.tsx

```diff
@@ -26,36 +64,77 @@ jest.mock('@context/server', () => ({
     useServerUrl: jest.fn().mockReturnValue('http://localhost:8065'),
 }));
 
+// Mock for custom profile attributes API
+const mockFetchCustomProfileAttributes = jest.fn();
+
+jest.mock('@actions/remote/custom_profile', () => ({
+    fetchCustomProfileAttributes: (...args: any[]) => mockFetchCustomProfileAttributes(...args),
+    updateCustomProfileAttributes: jest.fn().mockResolvedValue({success: true, error: undefined}),
+}));
+
 jest.mock('@actions/remote/user', () => ({
-    fetchCustomAttributes: jest.fn().mockResolvedValue({
-        attributes: {
-            attr1: {
+    updateMe: jest.fn().mockResolvedValue({error: undefined}),
+    uploadUserProfileImage: jest.fn().mockResolvedValue({error: undefined}),
+    setDefaultProfileImage: jest.fn().mockResolvedValue({}),
+    buildProfileImageUrlFromUser: jest.fn().mockReturnValue('http://example.com/profile.jpg'),
+}));
+
+// Mock WatermelonDB functionality
+jest.mock('@database/manager', () => {
+    const databaseMock = {
+        get: jest.fn().mockReturnValue({
+            query: jest.fn().mockReturnValue({
+                fetch: jest.fn().mockResolvedValue([]),
+            }),
+        }),
+    };
+
+    return {
+        getServerDatabaseAndOperator: jest.fn().mockReturnValue({
+            database: databaseMock,
+            operator: {
+                handleCustomProfileAttributes: jest.fn().mockResolvedValue({}),
+                handleCustomProfileFields: jest.fn().mockResolvedValue({}),
+            },
+        }),
+    };
+});
```

Comment:
any reason why we want to mock this?

I would defo not mock it and check the values have been updated in the db